rdt/transformers/utils.py

```diff
@@ -422,6 +425,10 @@ def _safe_parse_datetime(value, warn=False):
         return value
 
     try:
+        # Strings of large numbers cause parser.parse to overflow
+        if isinstance(value, str) and value.isdigit() and datetime_format:
+            return pd.to_datetime(value, format=datetime_format)
```

Comment:
```suggestion
        if isinstance(value, str) or value.isdigit() and datetime_format:
            value = pd.to_datetime(value, format=datetime_format)
```

We need the `value` below in order to detect the `timezone` for it. (We are either returning the `parser.parse` or `None`)

Reply:
parser.parse doesn't support datetime objects as input. If we want to handle timezones for this case, I would have to implement special logic for it.

Reply:
Can you confirm that for point 1 we want to set `self._has_multiple_timezones = False` when `'%z'` is in the `datetime_format`?

Reply:
@pvk-developer A test case using mixed timezone offsets fails if I add this change. The reason is that `_has_multiple_timezones` never gets set to True, since we are skipping that line. Could you clarify why this change is needed?

I implemented all your other recommendations, they are all good additions, this is the only one I'm unsure about.